modules/tools/src/main/java11/org/apache/ignite/tools/javadoc/IgniteLinkTaglet.java

```diff
@@ -69,65 +67,59 @@
         return true;
     }
 
-    /**
-     * Register this Taglet.
-     *
-     * @param tagletMap the map to register this tag to.
-     */
-    public static void register(Map<String, IgniteLinkTaglet> tagletMap) {
-        IgniteLinkTaglet tag = new IgniteLinkTaglet();
-
-        Taglet t = tagletMap.get(tag.getName());
-
-        if (t != null)
-            tagletMap.remove(tag.getName());
-
-        tagletMap.put(tag.getName(), tag);
-    }
-
     /**
      * Given the <code>Tag</code> representation of this custom tag, return its string representation.
      * <p>
      * Input: org.apache.ignite.grid.spi.indexing.h2.GridH2IndexingSpi#setIndexCustomFunctionClasses(Class[])
      * <p>
-     * Output: <a href="../../../../../org/apache/ignite/grid/spi/indexing/h2/GridH2IndexingSpi.html#
-     * setIndexCustomFunctionClasses(java.lang.Class...)">
-     * <code>GridH2IndexingSpi.setIndexCustomFunctionClasses(java.lang.Class[])</code></a>
+     * Output: &lt;a href="../../../../../org/apache/ignite/grid/spi/indexing/h2/GridH2IndexingSpi.html#
+     * setIndexCustomFunctionClasses(java.lang.Class...)"&gt;
+     * &lt;code&gt;GridH2IndexingSpi.setIndexCustomFunctionClasses(java.lang.Class[])&lt;/code&gt;&lt;/a&gt;
      *
-     * @param tag <code>Tag</code> representation of this custom tag.
+     * @param tags <code>DocTree</code> representation of this custom tag.
      */
-     public String toString(Tag tag) {
-        if (tag.text() == null || tag.text().isEmpty())
-            return "";
+    @Override public String toString(List<? extends DocTree> tags, Element element) {
+        for (DocTree tag : tags) {
+            String text = new SimpleDocTreeVisitor<String, Void>() {
+                @Override public String visitUnknownInlineTag(UnknownInlineTagTree node, Void param) {
+                    return node.getContent().toString();
+                }
+            }.visit(tag, null);
 
-        File f = tag.position().file();
+            if (text == null || text.isEmpty())
+                return "";
 
-        String curClass = f == null ? "" : f.getAbsolutePath().replace(File.separator, ".");
+            File f = new File(env.getDocTrees().getPath(element).getCompilationUnit().getSourceFile().toUri());
```

Comment:
This looks like a traverse tree operation ".getPath(element)".
Is there any reason to do this more than once?